### PR TITLE
Wip unified cdc

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-tram-messaging/src/main/java/io/eventuate/tram/messaging/common/sql/SqlDialectSelector.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    this.driver = driver;
+  }
+
+  @PostConstruct
+  private void init() {
+    sqlDialect = sqlDialects
+            .stream()
+            .filter(dialect -> dialect.supports(driver))
+            .min(OrderComparator.INSTANCE)
+            .orElseThrow(() -> new IllegalStateException(String.format("Sql Dialect not found (%s)", driver)));
```
Comment (cer) [inline eventuate-tram-messaging/src/main/java/io/eventuate/tram/messaging/common/sql/SqlDialectSelector.java]: This should error message should remind the user about the `eventuate.current.time.in.milliseconds.sql` property

File: eventuate-tram-messaging/src/test/java/io/eventuate/tram/messaging/common/sql/DefaultSqlDialectTest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = SqlDialectConfiguration.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
```
Comment (cer) [inline eventuate-tram-messaging/src/test/java/io/eventuate/tram/messaging/common/sql/DefaultSqlDialectTest.java]: Why is `@DirtiesContext` needed

File: eventuate-tram-messaging/src/test/resources/default.dialect.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+eventuate.current.time.in.milliseconds.sql=some custom sql
```
Comment (cer) [inline eventuate-tram-messaging/src/test/resources/default.dialect.properties]: You can replace all this test dialect.properties files with property definitions in the test `@SpringBootTest(properties = {"foo=bar", "bar=foo"})`

File: eventuate-tram-messaging/src/main/java/io/eventuate/tram/messaging/common/sql/SqlDialectSelector.java (hunk 1 above)
Comment (dartartem) [reply eventuate-tram-messaging/src/main/java/io/eventuate/tram/messaging/common/sql/SqlDialectSelector.java]: ok

File: eventuate-tram-messaging/src/test/java/io/eventuate/tram/messaging/common/sql/DefaultSqlDialectTest.java (hunk 2 above)
Comment (dartartem) [reply eventuate-tram-messaging/src/test/java/io/eventuate/tram/messaging/common/sql/DefaultSqlDialectTest.java]: it seems it is not necessary

File: eventuate-tram-messaging/src/test/resources/default.dialect.properties (hunk 3 above)
Comment (dartartem) [reply eventuate-tram-messaging/src/test/resources/default.dialect.properties]: ok

File: update-mysql.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+CREATE TABLE eventuate.offset_store(client_name VARCHAR(255) NOT NULL PRIMARY KEY, serialized_offset VARCHAR(255));
```
Comment (cer) [inline update-mysql.sql]: Please remove explicit schema name 'eventuate.' from script

File: update-postgres.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+CREATE TABLE eventuate.offset_store(client_name VARCHAR(255) NOT NULL PRIMARY KEY, serialized_offset VARCHAR(255));
```
Comment (cer) [inline update-postgres.sql]: Please remove explicit schema name 'eventuate.' from script

Comment (dartartem) [reply update-postgres.sql]: ok

File: update-mysql.sql (hunk 4 above)
Comment (dartartem) [reply update-mysql.sql]: ok

**Pull request merged**